core/qubes.py

```diff
@@ -194,6 +195,14 @@ def hvm_supported(self):
                     caps_xml.xpath('./guest/os_type[text()=\'hvm\']'))
         return self._hvm_supported
 
+    @property
+    def hap_supported(self):
+        if self._hap_supported is None:
+            caps_xml = lxml.etree.fromstring(self.libvirt_conn.getCapabilities())
+            self._hap_supported = bool(
+                    caps_xml.xpath('./guest/features/hap'))
```

Comment:
This isn't that simple - on non HAP capable system the hap node is still present: https://gist.github.com/marmarek/05a64b2d3dd23b6d6d35b87d2d008708